src/_guides/language/effective-dart/design.md

```diff
@@ -658,50 +696,121 @@ can.
 
 ### DO use getters for operations that conceptually access properties.
 
-If the name of the method starts with `get` or is a noun phrase like `length` or
-`size` that's a sign you're better off using a getter. You
-should define a getter instead of a method when all of these are true:
+Deciding when a member should be a getter versus a method is a challenging,
+subtle, but important part of good API design, hence this very long guideline.
+Some other language's cultures shy away from getters. They only use them when
+the operation is almost exactly like a field — it does a miniscule amount of
+calculation on state that lives entirely on the object. Anything more complex or
+heavyweight than that gets `()` after the name to signal "computation goin' on
+here!" because a bare name after a `.` means "field".
+
+Dart is *not* like that. In Dart, *all* dotted names are member invocations that
+may do computation. Fields are the special-case — getters whose implementation
+is provided by the language. In other words, getters are not "particularly slow
+fields" in Dart, fields are "particularly fast getters".
+
+Even so, choosing a getter over a method sends an important signal to the caller
+and you should use a getter when that's the signal you want to send. The signal,
+roughly, is that the operation is "field-like". The operation, at least in
+principle, *could* be implemented using a field, as far as the caller knows.
+That implies:
+
+*   **The operation does not take any arguments and returns a result.**
+
+*   **The caller cares mostly about the result.** If you want the caller to
+    worry about *how* the operation produces its result more than they do the
+    result being produced, then give the operation a verb name that describes
+    the work and make it a method.
+
+    This does *not* mean the operation has to be particularly fast in order to
+    be a getter. `IterableBase.length` is `O(n)`, and that's OK. It's fine for a
+    getter to do significant calculation. But if it does a *surprising* amount
+    of work, you may want to draw their attention to that by making it a method
+    whose name is a verb describing what it does.
+
+    {:.bad-style}
+    {% prettify dart %}
+    connection.nextIncomingMessage; // Does network IO.
+    expression.normalForm; // Could be exponential to calculate.
+    {% endprettify %}
 
-  * **Does not take any arguments.**
-  * **Returns a value.**
-  * **Is side-effect free.** Invoking a getter shouldn't change any
-  externally-visible state (caching internally or lazy initialization is
-  OK). Invoking the same getter repeatedly should return the same value
-  unless the object is explicitly changed between calls.
+*   **The operation does not have user-visible side effects.** Accessing a real
+    field does not alter the object or any other state in the program. It
+    doesn't produce output, write files, etc. A getter should generally follow
+    that.
```

Comment:
"that" doesn't seem like enough, but I don't know what else to use. "follow that precedent"? Maybe just delete the sentence?

Reply:
Deleting it feels like it leaves it up to the reader to make the connection. Tried a different sentence.